setup.py

```diff
@@ -17,22 +18,20 @@ def get_version(rel_path):
     raise RuntimeError("Unable to find version string.")
 
 
+def get_extra_deps(rel_path):
+    here = os.path.abspath(os.path.dirname(__file__))
+    extra_deps = json.load(open(os.path.join(here, rel_path), "r"))
+    return extra_deps
+
+
 # use techniques described at https://packaging.python.org/en/latest/guides/single-sourcing-package-version/
-# Don't use technique 6 since it need extra dependencies.
+# Don't use technique 6 since it needs extra dependencies.
```

Comment:
What is techinique 6?

Reply:
We cannot directly use version variable in __init__.py